File: frontend/lib/core/services/models/stats_response.dart
part of '../api_service.dart';

class StatsResponse {
  final List<dynamic> workouts;
  final int workoutCount;
  final double totalVolume;
  final int averageDurationMinutes;
  final Map<String, double> personalRecords;
  final Map<String, double> topExercisesByVolume;
  final String periodStart;
  final String periodEnd;

  StatsResponse({
    required this.workouts,
    required this.workoutCount,
    required this.totalVolume,
    required this.averageDurationMinutes,
    required this.personalRecords,
    required this.topExercisesByVolume,
    required this.periodStart,
    required this.periodEnd,
  });

  factory StatsResponse.fromJson(Map<String, dynamic> json) {
    return StatsResponse(
      workouts: json['workouts'] as List<dynamic>? ?? [],
      workoutCount: json['workoutCount'] as int? ?? 0,
      totalVolume: (json['totalVolume'] as num?)?.toDouble() ?? 0.0,
      averageDurationMinutes: json['averageDurationMinutes'] as int? ?? 0,
      personalRecords: _parseDoubleMap(json['personalRecords']),
      topExercisesByVolume: _parseDoubleMap(json['topExercisesByVolume']),
      periodStart: json['periodStart'] as String? ?? '',
      periodEnd: json['periodEnd'] as String? ?? '',
    );
  }

  static Map<String, double> _parseDoubleMap(dynamic data) {
    if (data == null) return {};
    if (data is Map) {
      return Map<String, double>.from(
        data.cast<String, dynamic>().map(
          (k, v) => MapEntry(k, (v as num).toDouble()),
        ),
      );
    }
    return {};
  }
}

File: backend/src/main/java/com/liftly/api/dto/StatsResponse.java
package com.liftly.api.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.util.List;
import java.util.Map;

/**
 * Comprehensive stats response for a user within a date range
 * Contains all data needed for stats page/dashboard
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class StatsResponse {
    // Workout data for charts
    private List<WorkoutResponse> workouts;
    
    // Summary metrics
    private Integer workoutCount;
    private Double totalVolume;
    private Integer averageDurationMinutes;
    
    // Rankings
    private Map<String, Double> personalRecords;
    private Map<String, Double> topExercisesByVolume;
    
    // Metadata
    private String periodStart;
    private String periodEnd;
}
